crates/relayer-cli/src/commands/evidence.rs

```diff
@@ -0,0 +1,786 @@
+    evidence: &LightClientAttackEvidence,
+    chain: &CosmosSdkChain,
+    counterparty_client: ForeignClient<BaseChainHandle, BaseChainHandle>,
+    counterparty_client_id: ClientId,
+    counterparty: &BaseChainHandle,
+    misbehaviour: TendermintMisbehaviour,
+) -> Result<(), eyre::Error> {
+    info!(
+        "building light client attack evidence for client `{}` on chain `{}`",
+        counterparty_client_id,
+        counterparty.id(),
+    );
+
+    let counterparty_is_provider = is_counterparty_provider(chain, counterparty);
+    let counterparty_client_is_frozen = counterparty_client.is_frozen();
+
+    if !counterparty_is_provider && counterparty_client_is_frozen {
+        warn!(
+            "cannot submit light client attack evidence to client `{}` on chain `{}`",
+            counterparty_client_id,
+            counterparty.id()
+        );
+        warn!("reason: client is frozen and chain is not a CCV provider chain");
+
+        return Ok(());
+    }
+
+    let signer = counterparty.get_signer()?;
+    let common_height = Height::from_tm(evidence.common_height, chain.id());
+
+    let provider_has_common_consensus_state = counterparty_is_provider
+        && has_consensus_state(counterparty, &counterparty_client_id, common_height);
+
+    if counterparty_is_provider
+        && counterparty_client_is_frozen
+        && !provider_has_common_consensus_state
+    {
+        warn!(
+            "cannot submit light client attack evidence to client `{}` on provider chain `{}`",
+            counterparty_client_id,
+            counterparty.id()
+        );
+        warn!("reason: client is frozen and does not have a consensus state at height {common_height}");
+
+        return Ok(());
+    }
+
+    let mut msgs = match counterparty_client.wait_and_build_update_client(common_height) {
+        Ok(msgs) => msgs,
+
+        Err(e) => {
+            warn!(
+                "skipping update client message for client `{}` on chain `{}`",
+                counterparty_client_id,
+                counterparty.id()
+            );
+            warn!("reason: failed to build update client message: {e}");
+
+            Vec::new()
+        }
+    };
+
+    if counterparty_is_provider {
+        info!(
+            "will submit consumer light client attack evidence to client `{}` on provider chain `{}`",
+            counterparty_client_id,
+            counterparty.id(),
+        );
+
+        let msg = MsgSubmitIcsConsumerMisbehaviour {
+            submitter: signer.clone(),
+            misbehaviour: misbehaviour.clone(),
+        }
+        .to_any();
+
+        msgs.push(msg);
+    };
+
+    // Only send the regular IBC message if the counterparty is not a provider,
+    // or if the counterparty is a provider but its client is not frozen already,
+    // and has a consensus state at the common height (implicity because of the check above).
+    if !counterparty_is_provider || !counterparty_client_is_frozen {
```

Comment:
Would this work?
```suggestion
    if !counterparty_client_is_frozen {
```

Reply:
Oh right, since the client is already frozen we don't need to submit that message, whether it's a provider chain or not.